src/epac/flatbuffers/pva0_data.py

```diff
@@ -0,0 +1,1002 @@
+        )
+    except KeyError:
+        raise ValueError(f"Unsupported scalar dtype: {data.dtype}")
+
+
+def serialise_any(builder: flatbuffers.Builder, data) -> int:
+    """serialises an arbitrary object into a FlatBuffers Any union.
+
+    Args:
+        builder (flatbuffers.Builder): The FlatBuffers builder used to construct the object.
+        data: The object to be serialised, which must be compatible with the Any union.
+
+    Returns:
+        The FlatBuffers offset for the serialised Any object.
+
+    Raises:
+        ValueError: If an attempt is made to serialise an empty array.
+    """
+    data = np.array(data)
+    dtype = data.dtype
+
+    # Handle empty arrays
+    if data.ndim > 0 and data.size == 0:
+        raise ValueError("Cannot serialise empty arrays.")
+
+    if data.ndim == 0:  # Scalar
+        if np.issubdtype(data.dtype, np.str_) or np.issubdtype(data.dtype, np.bytes_):
+            data_start_offset, fb_type, enum_type = _serialise_string_scalar(
+                builder, data
+            )
+        else:
+            data_start_offset, fb_type, enum_type = _serialise_numeric_scalar(
+                builder, data
+            )
+    else:  # Array
+        if np.issubdtype(dtype, np.str_) or np.issubdtype(dtype, np.bytes_):
+            data_start_offset, fb_type, enum_type = _serialise_string_array(
+                builder, data
+            )
+        else:
+            data_start_offset, fb_type, enum_type = _serialise_numeric_array(
+                builder, data
+            )
+
+    # Serialise the data using the FlatBuffer type
+    fb_type.Start(builder)
+    fb_type.AddValue(builder, data_start_offset)
+    data_offset = fb_type.End(builder)
+
+    AnyOuter.Start(builder)
+    AnyOuter.AddValueType(builder, enum_type)
+    AnyOuter.AddValue(builder, data_offset)
+    return AnyOuter.End(builder)
+
+
+def deserialise_any(buffer):
+    """Deserialises the Any table from a FlatBuffer.
+
+    Args:
+        buffer: FlatBuffer object containing the serialised Any table.
+
+    Returns:
+        The deserialised scalar or array value.
+
+    Raises:
+        ValueError: If the data type is unsupported.
+    """
+    data_enum = buffer.ValueType()
+
+    if data_enum in map_any_scalar_enum_to_type:
+        data_fb = map_any_scalar_enum_to_type[data_enum]()
+        data_offset = buffer.Value()
+        data_fb.Init(data_offset.Bytes, data_offset.Pos)
+        data = data_fb.Value()
+        if data_enum == AnyInner.AnyInner.String:
+            data = data.decode("utf-8")
+        return data
+
+    elif data_enum in map_any_array_enum_to_type:
+        data_fb = map_any_array_enum_to_type[data_enum]()
+        data_offset = buffer.Value()
+        data_fb.Init(data_offset.Bytes, data_offset.Pos)
+        if data_enum == AnyInner.AnyInner.StringArray:
+            data = np.array(
+                [str(data_fb.Value(n), "utf-8") for n in range(data_fb.ValueLength())]
+            )
```

Comment:
Ooh, one of my favourite tricks:

```suggestion
            data = np.fromiter(
                str(data_fb.Value(n), "utf-8") for n in range(data_fb.ValueLength()),
                str,  # This needs to be the numpy string dtype and I can't remember how it works
                data_fb.ValueLength(),  # Not necessary but I think this slightly helps performance
            )
```

Reply:
I would need to specify a max length apparently.